sycl/include/sycl/ext/oneapi/backend/level_zero.hpp

```diff
@@ -41,96 +41,29 @@
-           Interop) {
-  return make_platform(reinterpret_cast<pi_native_handle>(Interop));
-}
-
-// Construction of SYCL device.
-template <typename T,
-          typename std::enable_if_t<std::is_same_v<T, device>> * = nullptr>
-__SYCL_DEPRECATED("Use SYCL 2020 sycl::make_device free function")
-T make(const platform &Platform,
-       typename sycl::detail::interop<backend::ext_oneapi_level_zero, T>::type
-           Interop) {
-  return make_device(Platform, reinterpret_cast<pi_native_handle>(Interop));
-}
-
-/// Construction of SYCL context.
-/// \param DeviceList is a vector of devices which must be encapsulated by
-///        created SYCL context. Provided devices and native context handle must
-///        be associated with the same platform.
-/// \param Interop is a Level Zero native context handle.
-/// \param Ownership (optional) specifies who will assume ownership of the
-///        native context handle. Default is that SYCL RT does, so it destroys
-///        the native handle when the created SYCL object goes out of life.
-///
-template <typename T, std::enable_if_t<std::is_same_v<T, context>> * = nullptr>
-__SYCL_DEPRECATED("Use SYCL 2020 sycl::make_context free function")
-T make(const std::vector<device> &DeviceList,
-       typename sycl::detail::interop<backend::ext_oneapi_level_zero, T>::type
-           Interop,
-       ownership Ownership = ownership::transfer) {
-  return make_context(DeviceList,
-                      sycl::detail::pi::cast<pi_native_handle>(Interop),
-                      Ownership == ownership::keep);
-}
-
-// Construction of SYCL event.
-template <typename T,
-          typename std::enable_if_t<std::is_same_v<T, event>> * = nullptr>
-__SYCL_DEPRECATED("Use SYCL 2020 sycl::make_event free function")
-T make(const context &Context,
-       typename sycl::detail::interop<backend::ext_oneapi_level_zero, T>::type
-           Interop,
-       ownership Ownership = ownership::transfer) {
-  return make_event(Context, reinterpret_cast<pi_native_handle>(Interop),
-                    Ownership == ownership::keep);
-}
-
-} // namespace ext::oneapi::level_zero
-
 // Specialization of sycl::make_context for Level-Zero backend.
 template <>
 inline context make_context<backend::ext_oneapi_level_zero>(
     const backend_input_t<backend::ext_oneapi_level_zero, context>
         &BackendObject,
     const async_handler &Handler) {
-  (void)Handler;
-  return ext::oneapi::level_zero::make_context(
-      BackendObject.DeviceList,
-      detail::pi::cast<pi_native_handle>(BackendObject.NativeHandle),
-      BackendObject.Ownership == ext::oneapi::level_zero::ownership::keep);
+
+  const std::vector<device> &DeviceList = BackendObject.DeviceList;
+  pi_native_handle NativeHandle =
+      detail::pi::cast<pi_native_handle>(BackendObject.NativeHandle);
+  bool KeepOwnership =
+      BackendObject.Ownership == ext::oneapi::level_zero::ownership::keep;
+
+  return sycl::detail::make_context(NativeHandle, Handler,
```

Comment:
Previously `Handler` was ignored - see https://github.com/intel/llvm/blob/3ea29b2a9028b485b76339e16754e3e74c9cc7a6/sycl/source/backend/level_zero.cpp#L58-L60, this PR starts using it instead of `defaultAsyncHandler`.